src/qibo/models/dbi/double_bracket.py

```diff
@@ -20,6 +19,38 @@ class DoubleBracketGeneratorType(Enum):
     # TODO: add double commutator (does it converge?)
 
 
+class DoubleBracketCost(Enum):
+    """Define the DBI cost function."""
+
+    off_diagonal_norm = auto()
+    """Use off-diagonal norm as cost function."""
+    least_squares = auto()
+    """Use least squares as cost function."""
+    energy_fluctuation = auto()
+    """Use energy fluctuation as cost function."""
+
+
+from qibo.models.dbi.utils_scheduling import (
+    grid_search_step,
+    hyperopt_step,
+    polynomial_step,
+    simulated_annealing_step,
+)
```

Comment:
The issue lies on a circular import as qibo.models.dbi.utils_scheduling needs the DoubleBracketCost class (defined before) and double.bracket.py needs the step functions which are defined on qibo.models.dbi.utils_scheduling. I'm not too knowledgeable with how imports work so please let me know what would be the best solution (as you said it may be best to change the structure of the code)

Reply:
If there is a circular import there should be a way to disentangle the code. 
If I understood correctly the problem relies on the fact that `polynomial_step` depends on the cost function.
For a quick patch you could:
- remove the argument `cost` (so that we get rid of the type hint)
- when checking for the corresponding `coef` you could use strings instead of the instance of the class, to do this the `Enum` needs to inherit also from string
Something like this:
```py
from enum import Enum

class Type(str, Enum):
    a = "a"
    b = "b"

test = Type.a
test == "a' # returns True
```